homeassistant/components/ruckus_unleashed/coordinator.py

```diff
@@ -37,12 +31,15 @@ def __init__(self, hass: HomeAssistant, *, ruckus: Ruckus) -> None:
 
     async def _fetch_clients(self) -> dict:
         """Fetch clients from the API and format them."""
-        clients = await self.ruckus.current_active_clients()
-        return {e[API_MAC]: e for e in clients[API_CURRENT_ACTIVE_CLIENTS][API_CLIENTS]}
+        clients = await self.ruckus.api.get_active_clients()
+        _LOGGER.debug("fetched %d active clients", len(clients))
+        return {client[API_CLIENT_MAC]: client for client in clients}
 
     async def _async_update_data(self) -> dict:
         """Fetch Ruckus Unleashed data."""
         try:
-            return {API_CLIENTS: await self._fetch_clients()}
-        except (AuthenticationError, ConnectionError) as error:
-            raise UpdateFailed(error) from error
+            return {KEY_SYS_CLIENTS: await self._fetch_clients()}
+        except AuthenticationError as autherror:
+            raise UpdateFailed(autherror) from autherror
```

Comment:
We should probably raise `ConfigEntryAuthFailed` here.

Reply:
Done in https://github.com/home-assistant/core/pull/99411